src/hotspot/share/runtime/thread.hpp

```diff
@@ -1589,6 +1589,11 @@ inline JavaThread* JavaThread::current() {
   return Thread::current()->as_Java_thread();
 }
 
+inline WorkerThread* Thread::as_Worker_thread() const {
+  assert(is_Worker_thread(), "Dubious cast to WorkerThread*?");
+  return (WorkerThread*)this;
```

Comment:
Can you use static_cast here rather the C-style cast please.

Reply:
I get an compiling error, `static_cast from 'const Thread *' to 'WorkerThread *', which are not related by inheritance, is not allowed`. As `WorkerThread` is only declared in this context, the compiler doesn't know the inheritance relation here.